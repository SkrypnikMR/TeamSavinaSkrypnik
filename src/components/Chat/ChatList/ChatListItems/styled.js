import styled from 'styled-components';
import { bgColorDefault, colorDefault } from '/src/components/UI/baseLayout';

export const StChatListItems = styled.div`
    display: flex;
    width: 100%;
    height: 60px;
    align-items: center;
    color: ${({ color = colorDefault }) => color};
    cursor: pointer;
        ul {
          padding-left: 1rem;
          list-style-type: none;
        }
        :hover{
            color: ${({ color = bgColorDefault }) => color};
            span{
                color: ${({ color = colorDefault }) => color};
            }
        }
`;
export const StPhoto = styled.div`
    margin: 5px 10px;
    img{
        width: 50px;
        height: 50px;
    }
`;
export const StRoom = styled.div`
    display:flex;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
        p{
            width:80%;
        }
        span{
            width: 30px;
            height: 30px;
            display:flex;
            justify-content: center;
            align-items: center;
            font-size: 20px;
            border-radius: 50%;
            background-color: ${({ bgColor = bgColorDefault }) => bgColor};
        }
`;
