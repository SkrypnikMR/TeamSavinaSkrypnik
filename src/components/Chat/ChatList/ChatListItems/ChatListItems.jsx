import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { StChatListItems, StPhoto, StRoom } from './styled';
import { bgColorDefault } from '../../../UI/baseLayout';

const ChatListItems = ({
  id,
  img,
  color,
  content,
  setValue,
  unreadCount,
  resetUnreadCount,
  readAllMessagesInRoom,
  currentRoomName,
}) => {
  const handleClick = () => {
    setValue({ name: 'currentRoom', value: { room_id: id, room_name: content } });
    readAllMessagesInRoom({ room_id: id, room_name: content });
    resetUnreadCount(content);
  };
  useEffect(() => {
    if (currentRoomName === content) {
      return setState({ ...state, selected: true });
    }
    setState({ ...state, selected: false });
  }, [currentRoomName]);
  const [state, setState] = useState({
    error: false,
    src: img,
    defaultImg: './public/assets/images/defaultChats.png',
    selected: false,
  });
  const onError = () => setState({ ...state, error: true, src: state.defaultImg });
  return (
    <StChatListItems color={state.selected ? bgColorDefault : color} onClick={handleClick}>
      <StPhoto>
        <img src={img ? state.src : state.defaultImg} onError={onError} />
      </StPhoto>
      <StRoom>
        <p>{content}</p>
        {unreadCount ? <span>{unreadCount}</span> : null}
      </StRoom>
    </StChatListItems>
  );
};

ChatListItems.propTypes = {
  img: PropTypes.any,
  id: PropTypes.number,
  color: PropTypes.string,
  setValue: PropTypes.func,
  content: PropTypes.string,
  unreadCount: PropTypes.number,
  readAllMessagesInRoom: PropTypes.func.isRequired,
  resetUnreadCount: PropTypes.func.isRequired,
  currentRoomName: PropTypes.string,
};

export default ChatListItems;
