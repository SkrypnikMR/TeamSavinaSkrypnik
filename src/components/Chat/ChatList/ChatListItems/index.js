import { connect } from 'react-redux';
import ChatListItems from './ChatListItems.jsx';
import { setValue, readAllMessagesInRoom, resetUnreadCount } from '/src/Store/chat/actions';
import { currentRoomName } from '/src/Store/chat/selectors';

const mapStateToProps = state => ({
    currentRoomName: currentRoomName(state),
});
const mapDispatchToProps = dispatch => ({
    setValue: payload => dispatch(setValue(payload)),
    readAllMessagesInRoom: payload => dispatch(readAllMessagesInRoom(payload)),
    resetUnreadCount: payload => dispatch(resetUnreadCount(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatListItems);
