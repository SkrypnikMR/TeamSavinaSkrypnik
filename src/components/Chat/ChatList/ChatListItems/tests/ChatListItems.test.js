import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import ChatListItems from '../ChatListItems';

describe('ChatListItems', () => {
  let props;
  beforeEach(() => {
    props = {
      content: '',
      img: '',
      color: '',
    };
  });
  it('Should match snapshot', () => {
    const component = shallow(<ChatListItems {...props} />);
    expect(component.html).toMatchSnapshot();
  });
  it('Should render StPhoto', () => {
    const component = shallow(<ChatListItems {...props} />);
    expect(component.find('styled__StPhoto')).toHaveLength(1);
  });
  it('Should render img', () => {
    const component = shallow(<ChatListItems {...props} />);
    expect(component.find('img')).toHaveLength(1);
  });
  it('Should render div', () => {
    const component = shallow(<ChatListItems {...props} />);
    expect(component.find('div')).toHaveLength(1);
  });
  it('Should render defaultImg', () => {
    const props = {
      content: '',
      img: 'wrongImagePath',
      color: '',
    };
    act(() => {
      const component = mount(<ChatListItems {...props} />);
      const onError = component.find('img').prop('onError');
      const image = component.find('img');
      onError();
      expect(image.prop('src')).toBe(props.img);
    });
  });
});
