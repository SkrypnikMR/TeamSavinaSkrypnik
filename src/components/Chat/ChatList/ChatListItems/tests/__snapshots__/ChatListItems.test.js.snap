// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatListItems Should match snapshot 1`] = `[Function]`;
